wifiphisher/common/interfaces.py

```diff
@@ -243,6 +242,40 @@ def up_ifaces(self, ifaces):
         for i in ifaces:
             card = pyw.getcard(i.get_name())
             pyw.up(card)
+    
+    def change_ifaces_mac(self, ifaces, mac_addrs):
+        """
+        Randomize the MACs for the network adapters
+
+        :param self: A NetworkManager object
+        :param ifaces: A list of NetworkAdapter objects
+        :param mac_addrs: A list of MAC address
+        :type self: NetworkManager
+        :type ifaces: List
+        :return: None
+        :rtype: None
+        """
+
+        for i in range(len(ifaces)):
+            adapter = ifaces[i]
+            card = pyw.getcard(adapter.get_name())
+            is_up = pyw.isup(card)
+            mode = pyw.modeget(card)
+            #before change the mac address power down card
+            pyw.down(card)
+
+            if mode != 'managed':
+                pyw.modeset(card, 'managed')
+
+            pyw.macset(card, mac_addrs[i])
```

Comment:
I didn't see anything in `PyRIC` requiring the device to be in `managed` mode. Even if you want to do it once should be enough after setting the MAC address.